### PR TITLE
Fixes xesmf flag

### LINKED ISSUE #339: [Bug]: Support regridding for curvilinear grid
### Is your feature request related to a problem?

CDAT supported regridding for curvilinear grids, but this does not appear to work in xCDAT.

```python
import cdms2
import numpy as np

# open file
fn = '/p/css03/cmip5_css01/data/cmip5/output1/CNRM-CERFACS/CNRM-CM5-2/historical/mon/ocean/Omon/r1i1p1/v20130401/thetao/thetao_Omon_CNRM-CM5-2_historical_r1i1p1_185001-185912.nc'
f = cdms2.open(fn)
thetao = f('thetao')
f.close()
# define new grid
gridOut = createGeoGrid(np.arange(-88.5, 90, 2.5), np.arange(1.25, 360, 2.5))
# do regridding
diag = {}
thetaor = field.thetao(gridOut, regridTool='esmf', regridMethod='linear', missing=np.nan, coordSys='deg', diag=diag, periodicity=1)
thetaor.getAxisIds()
thetaor.shape
```

> ['time', 'lev', 'latitude', 'longitude'] 
> (120, 42, 72, 144)

```python
import xcdat as xc
from xcdat.regridder import grid
import numpy as np

nlat = np.arange(-88.75, 90, 2.5)
nlon = np.arange(1.25, 360, 2.5)
ngrid = grid.create_grid(nlat, nlon)
fn = '/p/css03/cmip5_css01/data/cmip5/output1/CNRM-CERFACS/CNRM-CM5-2/historical/mon/ocean/Omon/r1i1p1/v20130401/thetao/thetao_Omon_CNRM-CM5-2_historical_r1i1p1_185001-185912.nc'
ds = xc.open_dataset(fn)
print(ds)
ds = ds.regridder.horizontal('thetao', ngrid, tool='xesmf', method='bilinear', periodic=True)
```

> <xarray.Dataset>                                                                                                                                            
> Dimensions:       (time: 120, bnds: 2, lev: 42, j: 292, i: 362, vertices: 4)
> Coordinates:
>   * time          (time) datetime64[ns] 1850-01-16T12:00:00 ... 1859-12-16T12...
>   * lev           (lev) float64 5.022 15.08 25.16 ... 5.051e+03 5.35e+03
>   * j             (j) int32 1 2 3 4 5 6 7 8 ... 285 286 287 288 289 290 291 292
>   * i             (i) int32 1 2 3 4 5 6 7 8 ... 355 356 357 358 359 360 361 362
>     lat           (j, i) float32 -78.39 -78.39 -78.39 ... 50.23 50.01 50.01
>     lon           (j, i) float32 72.5 73.5 74.5 75.5 ... 73.05 73.04 73.0 72.99
> Dimensions without coordinates: bnds, vertices
> Data variables:
>     time_bnds     (time, bnds) datetime64[ns] ...
>     lev_bnds      (lev, bnds) float64 ...
>     lat_vertices  (j, i, vertices) float32 ...
>     lon_vertices  (j, i, vertices) float32 ...
>     thetao        (time, lev, j, i) float32 ...
> Attributes: (12/28)
>     institution:            CNRM (Centre National de Recherches Meteorologiqu...
>     institute_id:           CNRM-CERFACS
>     experiment_id:          historical
>     source:                 CNRM-CM5-2 2010 Atmosphere: ARPEGE-Climat (V5.2.3...
>     model_id:               CNRM-CM5-2
>     forcing:                GHG, SA, Sl, Vl, BC, OC
>     ...                     ...
>     table_id:               Table Omon (26 July 2011) e26ed05e72b60ab734809dc...
>     title:                  CNRM-CM5-2 model output prepared for CMIP5 histor...
>     parent_experiment:      pre-industrial control
>     modeling_realm:         ocean
>     realization:            1
>     cmor_version:           2.7.1

> ValueError: ESMC_FieldRegridStore failed with rc = 506. Please check the log files (named "*ESMF_LogFile").

### Describe the solution you'd like

Ideally, xcdat could recognize this grid and have the capability to regrid it. 

### Describe alternatives you've considered

_No response_

### Additional context

Actually – this was tested (see [this thread](https://github.com/xCDAT/xcdat/pull/164#issuecomment-1151693095)). I'm having issues in this issue here and the dirst sea ice dataset I pulled up (/p/css03/cmip5_css01/data/cmip5/output1/CNRM-CERFACS/CNRM-CM5-2/historical/mon/seaIce/OImon/r1i1p1/v20130401/sic/).

### PR DESCRIPTION
**Pull request by jasonb5**
## Description
Fixes `ignore_degenerate` not being passed through to xESMF regridder object.

- Closes #339 

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules

If applicable:

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass with my changes (locally and CI/CD build)
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have noted that this is a breaking change for a major release (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

**Comment by jasonb5** [comment]
This should resolve issues with curvilinear grids not working (#339).

File: tests/test_regrid.py [hunk 1]
```diff
@@ -438,6 +438,26 @@ def test_regrid(self):
...
+        "name,value,attr_name",
+        [
+            ("periodic", True, "_periodic"),
+            ("extrap_method", "inverse_dist", "_extrap_method"),
+            ("extrap_method", "nearest_s2d", "_extrap_method"),
+            ("extrap_dist_exponent", 0.1, "_extrap_dist_exponent"),
+            ("extrap_num_src_pnts", 10, "_extrap_num_src_pnts"),
+            ("ignore_degenerate", False, "_ignore_degenerate"),
+        ],
+    )
```
Comment (tomvothecoder) [inline tests/test_regrid.py:451]: I didn't know this pytest decorator exists. This is neat for easily testing different function/method parameters!

**Review by tomvothecoder — APPROVED** [review]
LGTM as far as code changes. @pochedls can you test out the functionality is working as expected?

**Review by pochedls — APPROVED** [review]
This resolves most of the curvilinear gridding issues. I do not have an exact match with CDAT, but when I spot-checked the grid cell differences were order 10$^{-5}$. In looking at 58 models, 9 had issues, but they appeared to be known issues (I think compared to more than 30 exceptions before?): 

| Error | Instances |
| --- | --- |
| `TypeError: buffer is too small for requested array` | 1 |
| `KeyError: "Receive multiple variables for key 'longitude': {'lon', 'longitude'}. Expected only one. Please pass a list ['longitude'] instead to get all variables matching 'longitude'."` | 2 |
| `KeyError: 'lon_b'` | 3 | 
| `ValueError: Bounds format not understood. Got ('nj', 'ni', 'vertices') with shape (384, 320, 2).` | 1 | 
| `KeyError: "No results found for 'longitude'."` | 1 | 
| `xarray.core.merge.MergeError: conflicting values for variable 'vertices_latitude' on objects to be combined. You can skip this check by specifying compat='override'.` | 1 | 

I think this PR is good to go. Thanks for sorting this out @jasonb5!

**Pull request merged**